addons/l10n_in_ewaybill_stock/models/l10n_in_ewaybill.py

```diff
@@ -0,0 +1,588 @@
+            else:
+                ewaybill.partner_bill_to_id = picking_id.partner_id
+                ewaybill.partner_bill_from_id = picking_id.company_id.partner_id
+                ewaybill.partner_ship_to_id = picking_id.partner_id
+                ewaybill.partner_ship_from_id = picking_id.picking_type_id.warehouse_id.partner_id
+                if partner_invoice_id := ewaybill.picking_id._l10n_in_get_invoice_partner():
+                    ewaybill.partner_bill_to_id = partner_invoice_id
+
+            if (
+                ewaybill.picking_type_code == 'dropship' and
+                (dest_partner := ewaybill.picking_id._get_l10n_in_dropship_dest_partner())
+            ):
+                ewaybill.partner_ship_to_id = dest_partner
+                ewaybill.partner_ship_from_id = ewaybill.picking_id.partner_id
+
+    def _compute_is_fiscal_position_invisible(self):
+        for ewaybill in self:
+            fiscal_position = ewaybill.picking_id._l10n_in_get_fiscal_position()
+            ewaybill.is_fiscal_position_invisible = (
+                fiscal_position == self.env['account.fiscal.position'] or
+                fiscal_position
+            )
+
+    @api.depends('partner_bill_from_id', 'partner_bill_to_id')
+    def _compute_fiscal_position(self):
+        for ewaybill in self.filtered(lambda ewb: ewb.state == 'pending'):
+            ewaybill.fiscal_position_id = (
+                ewaybill.picking_id._l10n_in_get_fiscal_position() or
+                self.env['account.fiscal.position']._get_fiscal_position(
+                    ewaybill.picking_type_code == 'incoming' and ewaybill.partner_bill_from_id or
+                    ewaybill.partner_bill_to_id
+                )
+            )
+
+    @api.depends('partner_ship_from_id', 'partner_ship_to_id', 'partner_bill_from_id', 'partner_bill_to_id')
+    def _compute_is_editable(self):
+        for ewaybill in self:
+            is_incoming = ewaybill.picking_type_code == "incoming"
+            ewaybill.is_bill_to_editable = not is_incoming
+            ewaybill.is_bill_from_editable = is_incoming
+            ewaybill.is_ship_from_editable = is_incoming and ewaybill._is_overseas()
+            ewaybill.is_ship_to_editable = not is_incoming and not ewaybill._is_overseas()
+
+    def _compute_content(self):
+        for ewaybill in self:
+            ewaybill.content = base64.b64encode(json.dumps(ewaybill._ewaybill_generate_direct_json()).encode())
+
+    @api.depends('name', 'state')
+    def _compute_display_name(self):
+        for ewaybill in self:
+            ewaybill.display_name = ewaybill.state == 'pending' and _('Pending') or ewaybill.name
+
+    @api.depends('mode')
+    def _compute_vehicle_type(self):
+        """when transportation mode is ship then vehicle type should be Over Dimensional Cargo (ODC)"""
+        for ewaybill in self.filtered(lambda ewb: ewb.state == 'pending' and ewb.mode == "4"):
+            ewaybill.vehicle_type = 'O'
```

Comment:
and the others should be False then?  (you did not have a Cachemiss?)

Reply:
Stored field. Not required.